test/Parse/diagnose_availability.swift

```diff
@@ -0,0 +1,27 @@
+// RUN: %target-typecheck-verify-swift 
+
+// SR-4231: Misleading/wrong error message for malformed @available
+
+@available(OSX 10.6, *) // no error
+func availableSince10_6() {}
+
+@available(OSX, introduced: 10.0, deprecated: 10.12) // no error
+func introducedFollowedByDepreaction() {}
```

Comment:
Another nit: here and below (not that it matters for your test), typo "Depreaction".